model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/JpaMapStorageProviderFactory.java

```diff
@@ -259,6 +259,7 @@ public MapStorageProvider create(KeycloakSession session) {
         if (provider == null) {
             provider = new JpaMapStorageProvider(this, session, getEntityManager(), this.sessionTxKey);
             session.setAttribute(this.sessionProviderKey, provider);
+            session.enlistForClose(provider);
```

Comment:
Good catch. This needs to be fixed elsewhere, see #14730.

Reply:
Noted, rebased